libass/ass_shaper.c

```diff
@@ -271,7 +271,7 @@ get_reference_table(hb_face_t *hbface, hb_tag_t tag, void *font_data)
   }
 
   hb_blob_t *blob = hb_blob_create(buf, len, HB_MEMORY_MODE_WRITABLE, buf, free);
-  if (!blob)
+  if (!blob || (len > 0 && hb_blob_is_immutable(blob)))
```

Comment:
afaik HarfBuzz _never_ returns `NULL` but always statically allocated dummy objects making the `!blob` check redundant

Reply:
I'm inclined to keep the checks since that behavior isn't actually documented anywhere afaik; it's not like this is hot-path code.

Reply:
it’s a conscious design decision; all API which can return NULL ends with `_or_fail` and explicitly documents a `NULL` return is possible, see e.g. [hb_subset_input_create_or_fail](https://harfbuzz.github.io/harfbuzz-hb-subset.html#hb-subset-input-create-or-fail)

While `hb_bloc_create` is documented to return a new blob (and nothing else, i.e. no `NULL`)

Reply:
Alrighty, fair enough; done.